src/prpy/serialization.py

```diff
@@ -327,37 +344,66 @@ def deserialize_kinbody(env, data, name=None, anonymous=False, state=True):
         'Robot' if data['is_robot'] else 'KinBody',
         data['name']
     )
-
-    link_infos = [
-        deserialize_link_info(link_data['info']) \
-        for link_data in data['links']
-    ]
-    joint_infos = [
-        deserialize_joint_info(joint_data['info']) \
-        for joint_data in data['joints']
-    ]
-
-    if data['is_robot']:
-        # TODO: Also load sensors.
-        manipulator_infos = [
-            deserialize_manipulator_info(manipulator_data['info']) \
-            for manipulator_data in data['manipulators']
-        ]
-        sensor_infos = []
-
-        kinbody = RaveCreateRobot(env, '')
-        kinbody.Init(
-            link_infos, joint_infos,
-            manipulator_infos, sensor_infos,
-            data['uri']
-        )
+    
+    name_desired = name or data['name']
+    
+    if 'uri_only' in data and data['uri_only']:
+        
+        if data['is_robot']:
+            
+            if data['uri'].endswith('.robot.xml'):
+                kinbody = env.ReadRobotXMLFile('robots/barrettwam.robot.xml')
```

Comment:
Ugh, I *guess* I can remove this.  But seriously, when was the last time that you wanted to load an XML file and it wasn't the WAM?  I think it's actually more likely that you wanted to load the WAM but got the path wrong (which this will fix!) than that you want to load some other robot. `(-:`